flow/connectors/mysql/cdc.go

```diff
@@ -601,3 +623,113 @@ func QValueFromMysqlRowEvent(mytype byte, qkind qvalue.QValueKind, val any) (qva
 	}
 	return nil, fmt.Errorf("unexpected type %T for mysql type %d", val, mytype)
 }
+
+func (c *MySqlConnector) buildTableSchemaLookup(req *model.PullRecordsRequest[model.RecordItems]) error {
+	if c.tableSchemaLookup == nil {
+		c.tableSchemaLookup = make(map[string][]string, len(req.TableNameMapping))
+		for sourceTableName := range maps.Keys(req.TableNameMapping) {
+			schemaTable, err := utils.ParseSchemaTable(sourceTableName)
+			if err != nil {
+				return err
+			}
+			c.tableSchemaLookup[schemaTable.Table] = append(c.tableSchemaLookup[schemaTable.Table], schemaTable.Schema)
```

Comment:
this breaks when table name is duplicated across schemas, which multitenant apps are want to do